backend/services/mgmt/v1alpha1/connection-service/connection.go

```diff
@@ -39,17 +39,99 @@ func (s *Service) CheckConnectionConfig(
 
 	cctx, cancel := context.WithTimeout(ctx, 5*time.Second)
 	defer cancel()
-	err = db.PingContext(cctx)
+	// err = db.PingContext(cctx)
+	tx, err := db.BeginTx(cctx, &sql.TxOptions{ReadOnly: true})
 	if err != nil {
-		msg := err.Error()
+		return nil, err
+	}
+	defer nucleusdb.HandleSqlRollback(tx, logger)
+
+	// queries the database to see if we can see the schemas and tables
+	// note that schemas with no tables will not appear in this query
+	pgQuery := `
+	SELECT 
+		nspname AS schema_name, 
+		COUNT(*) AS number_of_tables
+	FROM 
+		pg_catalog.pg_tables
+		JOIN pg_catalog.pg_namespace ON pg_namespace.nspname = pg_tables.schemaname
+	WHERE 
+		pg_namespace.oid NOT IN 
+			(SELECT oid FROM pg_catalog.pg_namespace WHERE nspname = 'pg_catalog' OR nspname = 'information_schema')
```

Comment:
I think this query can be reduced down to a single table.
```
SELECT 
  schemaname, 
  COUNT(*) AS number_of_tables
FROM 
  pg_catalog.pg_tables
WHERE
  schemaname NOT IN ('pg_catalog', 'information_schema')
GROUP BY schemaname;
```

Reply:
Also I don't know if the order by is really necessary unless you are using it to display more nicely on the frontend.